numpy/core/tests/test_umath.py

```diff
@@ -1840,6 +1840,24 @@ def test_reduction(self):
             btype = np.array([True], dtype=object)
             assert_(type(f.reduce(btype)) is bool, msg)
 
+    @pytest.mark.parametrize("input_dtype",
+        np.sctypes["int"] + np.sctypes["uint"])
```

Comment:
```suggestion
    @pytest.mark.parametrize("input_dtype", np.typecodes["AllInteger"] + ["?"])
```
Suggestion (feel free to ignore): using `np.typecodes` allows you to use a 1-liner here.

Also, adding the bool dtype to the test-list would be a good idea.

Reply:
Ohh nice, thanks @BvB93 , will add it.

Reply:
Hey @BvB93 , just wanted to know if there is a neat way to get say `np.int32` from `i` or `np.int64` from `l`. I tried `map(np.dtype, np.typecodes["AllInteger"] + "?")`, but this gives dtype object not the integer class.
Not an issue actually, but wanted to know if there is a nice way to do it :)

Reply:
Sure, you can get the corresponding scalar type via, for example, `np.dtype("i").type`.

Reply:
Ohh I just got it, I did `map(lambda c: np.dtype(c).type, np.typecodes["AllInteger"] + "?")`

Reply:
refined in 81628e7